# Notebook 01 - Inspección Inicial

En este notebook realizamos una primera exploración del dataset original 
sin modificar ningún dato. El objetivo es entender la estructura general, 
identificar problemas de calidad y formular preguntas de análisis.

In [1]:
# Importamos las librerías necesarias para la inspección
import pandas as pd
import json

In [2]:
# Cargamos el dataset original desde la carpeta raw (sin modificarlo)
with open('../data/raw/streaming_users_dirty.json') as f:
    data = json.load(f)

df = pd.DataFrame(data)
print("Dataset cargado correctamente.")

Dataset cargado correctamente.


In [3]:
# Verificamos cuántas filas y columnas tiene el dataset
print(f"El dataset tiene {df.shape[0]} filas y {df.shape[1]} columnas.")

El dataset tiene 8160 filas y 8 columnas.


In [4]:
# Observamos los primeros registros para entender qué representa cada columna
df.head()

,user_id,age,subscription_plan,monthly_watch_time_mins,country,favorite_genre,last_login_date,customer_support_tickets
0,10000,39,Estándar,805.8,Brasil,Crime,2025-03-04,99
1,10001,37,Estándar,1173.4,Colombia,Crime,2019-04-02,2
2,10002,28,Básico,401.0,Colombia,Crime,2018-04-13,0
3,10003,43,Básico,62.4,Uruguay,Thriller,2021-01-31,0
4,10004,51,Básico,477.8,Perú,Thriller,2020-09-30,1


In [4]:
# Revisamos el tipo de dato de cada columna
# Esto nos indica si las fechas, números y textos están bien clasificados
print("=== TIPOS DE DATOS ===")
print(df.dtypes)

=== TIPOS DE DATOS ===
user_id                       int64
age                           int64
subscription_plan               str
monthly_watch_time_mins     float64
country                         str
favorite_genre                  str
last_login_date                 str
customer_support_tickets      int64
dtype: object


In [5]:
# Contamos cuántos valores faltantes hay en cada columna
print("=== VALORES NULOS POR COLUMNA ===")
print(df.isnull().sum())
print()
print(f"Total de valores nulos: {df.isnull().sum().sum()}")

=== VALORES NULOS POR COLUMNA ===
user_id                       0
age                           0
subscription_plan             0
monthly_watch_time_mins     193
country                       0
favorite_genre              240
last_login_date             320
customer_support_tickets      0
dtype: int64

Total de valores nulos: 753


In [6]:
# Verificamos si hay filas duplicadas en el dataset
print(f"Filas duplicadas encontradas: {df.duplicated().sum()}")

Filas duplicadas encontradas: 126


In [7]:
# Resumen estadístico de todas las columnas
# Nos permite detectar valores mínimos y máximos inusuales
df.describe(include='all')

,user_id,age,subscription_plan,monthly_watch_time_mins,country,favorite_genre,last_login_date,customer_support_tickets
count,8160.000000,8160.000000,8160,7967.000000,8160,7920,7840,8160.000000
unique,NaN,NaN,15,NaN,26,28,3062,NaN
top,NaN,NaN,Básico,NaN,Brasil,Comedia,2026-15-03,NaN
freq,NaN,NaN,3450,NaN,1132,1112,20,NaN
mean,13995.433824,34.096814,NaN,1107.346153,NaN,NaN,NaN,1.800980
std,2310.810660,14.511304,NaN,5310.442604,NaN,NaN,NaN,11.334969
min,10000.000000,-5.000000,NaN,-120.000000,NaN,NaN,NaN,-1.000000
25%,11987.750000,25.000000,NaN,489.200000,NaN,NaN,NaN,0.000000
50%,13998.500000,33.000000,NaN,757.400000,NaN,NaN,NaN,1.000000
75%,15997.250000,42.000000,NaN,1045.700000,NaN,NaN,NaN,1.000000


In [8]:
# Revisamos cuántos valores únicos tiene cada columna
# Útil para detectar inconsistencias en variables categóricas
for col in df.columns:
    print(f"{col}: {df[col].nunique()} valores únicos")

user_id: 8000 valores únicos
age: 69 valores únicos
subscription_plan: 15 valores únicos
monthly_watch_time_mins: 5788 valores únicos
country: 26 valores únicos
favorite_genre: 28 valores únicos
last_login_date: 3062 valores únicos
customer_support_tickets: 9 valores únicos


## Observaciones iniciales

Durante la inspección se identificaron los siguientes problemas de calidad
que deberán abordarse en la etapa de limpieza:

 **Valores nulos**: se detectaron nulos en `monthly_watch_time_mins` (193),
  `favorite_genre` (240) y `last_login_date` (320).

 **Filas duplicadas**: se encontraron 126 filas duplicadas.

 **Variables categóricas inconsistentes**: `subscription_plan`, `country` y
  `favorite_genre` presentan múltiples variantes para un mismo valor
  (mayúsculas, minúsculas, abreviaturas, errores tipográficos).

 **Valores imposibles**: `age` tiene valores negativos y superiores a 100.
  `customer_support_tickets` tiene valores negativos y outliers extremos (99, 150).

 **Fechas mal formateadas**: `last_login_date` está como texto y contiene
  fechas inválidas como "2026-15-03".

 **Outliers en variables numéricas**: `monthly_watch_time_mins` tiene valores
  negativos y un máximo de 99999, que no es realista.

## Preguntas de análisis

A partir de la inspección del dataset, definimos las siguientes preguntas
que orientarán el análisis:

1. ¿El plan de suscripción influye en la cantidad de horas que un usuario
   consume contenido mensualmente?
2. ¿Existe alguna preferencia de género según el país de origen del usuario?
3. ¿A mayor edad, menor tiempo de visualización mensual?
4. ¿Los usuarios que generan más tickets de soporte son los que menos
   tiempo pasan en la plataforma?